### PR TITLE
fix: unpin maplibre-gl-js-amplify

### LINKED ISSUE #4142: MapView not loading after splitting into `@aws-amplify/react-ui-geo` (request transformer error)
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Geo (MapView)

### How is your app built?

Vite

### What browsers are you seeing the problem on?

Chrome, Firefox, Safari

### Please describe your bug.

I am trying to display a `MapView` using the new-ish `@aws-amplify/react-ui-geo` package and I'm not able to. Looking at the logs (see below) it seems that the request transformer from the Amplify Maplibre module is not handling/reading some configs properly, causing the request to never be sent.

I can confirm that the same exact code (except for the import change) works with older versions of Amplify UI (up until it was broken down into a separate package).

### What's the expected behaviour?

The `MapView` should render normally.

### Help us reproduce the bug!

You can use this minimal repro sample I have created: https://github.com/dreamorosi/amplify-ui-issue

If not:
- create a blank Vite app
- then install `aws-amplify`, `@aws-amplify/react-ui`, and `@aws-amplify/react-ui-geo` (all latest)
- create the map resources (or copy an existing `aws-exports.js` in the `src` folder
- try adding a map (see code below)
- start the app & observe error in the console + blank screen

### Code Snippet

```js
// Put your code below this line.
import React from "react";
import { MapView } from "@aws-amplify/ui-react-geo";
import { NavigationControl } from "react-map-gl";

const App = () => {
  return (
    <MapView
      initialViewState={{
        longitude: -115.17077150978058,
        latitude: 36.12309017212961,
        zoom: 15,
      }}
      style={{
        width: "100vw",
        height: "100vh",
      }}
    >
      <NavigationControl position={"top-left"} />
    </MapView>
  );
};


export default App;
```

### Additional information and screenshots

image

Logs:

```
[vite] connecting...
[vite] connected.
react_devtools_backend_compact.js:2367 TypeError: str.replace is not a function
    at urlEncodePeriods (utils.js:62:16)
    at AmplifyMapLibreRequest.transformRequest [as _transformRequestFn] (AmplifyMapLibreRequest.js:80:86)
    at r2.transformRequest (request_manager.ts:24:25)
    at ie.loadURL (style.ts:220:50)
    at Map._updateStyle (map.ts:1442:24)
    at Map.setStyle (map.ts:1395:25)
    at new Map (map.ts:464:33)
    at Mapbox._initialize (mapbox.ts:546:22)
    at new Mapbox (mapbox.ts:438:10)
    at map.tsx:103:22
```

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

Unpin `maplibre-gl-js-amplify` since it is an internal package and bump to the latest.

#### Issue #, if available

fixes #4142 

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changeset/ten-crabs-travel.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@aws-amplify/ui-react-geo": patch
+"@aws-amplify/ui-react": patch
```
Comment (calebpollman) [inline .changeset/ten-crabs-travel.md:3]: This doesn't impact ui-react i think?

Comment (reesscot) [reply .changeset/ten-crabs-travel.md:3]: +1

Comment (zchenwei) [reply .changeset/ten-crabs-travel.md:3]: The changeset automatically generates it

Comment (zchenwei) [reply .changeset/ten-crabs-travel.md:3]: idk why let me remove it

Comment (zchenwei) [reply .changeset/ten-crabs-travel.md]: done

**Review by calebpollman — APPROVED** [review]

File: .changeset/ten-crabs-travel.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@aws-amplify/ui-react-geo": patch
+---
+
+fix: unpin maplibre-gl-js-amplify
```
Comment (reesscot) [inline .changeset/ten-crabs-travel.md:5]: should we reference the issue we are fixing here?

Comment (zchenwei) [reply .changeset/ten-crabs-travel.md:5]: good point

**Review by reesscot — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**